### PR TITLE
Use logging library from go-kit

### PR DESCRIPTION
**Pull request by bharaththiruveedula**

### CODE REVIEW

**Comment by bharaththiruveedula** [comment]
@matthiasr, can you please review this?

**Comment by matthiasr** [comment]
To make the tests pass, please `go mod vendor` and commit the updates.

File: main.go [hunk 1]
```diff
@@ -72,7 +75,7 @@ func (c *influxDBCollector) serveUdp() {
 	for {
 		n, _, err := c.conn.ReadFromUDP(buf)
 		if err != nil {
-			log.Warnf("Failed to read UDP message: %s", err)
+			level.Warn(c.logger).Log("msg", "Failed to read UDP message", "warn", err)
```
Comment (matthiasr) [inline main.go]: is this the way it's supposed to be done in go-kit? We are already logging at warning level here, so I don't think we need a "warn" key to tell us that again? I'd rather use `error` as in the other log lines:
```suggestion
			level.Warn(c.logger).Log("msg", "Failed to read UDP message", "error", err)
```

File: main.go [hunk 2]
```diff
@@ -300,6 +306,8 @@ func main() {
     </html>`))
 	})
 
-	log.Infoln("Listening on", *listenAddress)
-	http.ListenAndServe(*listenAddress, nil)
+	if err := http.ListenAndServe(*listenAddress, nil); err != nil {
+		level.Error(logger).Log("msg", "Error starting HTTP server", "err", err)
```
Comment (matthiasr) [inline main.go]: for consistency

```suggestion
		level.Error(logger).Log("msg", "Error starting HTTP server", "error", err)
```

File: main.go (hunk 1 above)
Comment (brian-brazil) [reply main.go]: `err` is the usual key.

Comment (matthiasr) [reply main.go]: ah! updated the suggestion

File: main.go [hunk 3]
```diff
@@ -82,7 +85,7 @@ func (c *influxDBCollector) serveUdp() {
 		precision := "ns"
 		points, err := models.ParsePointsWithPrecision(bufCopy, time.Now().UTC(), precision)
 		if err != nil {
-			log.Errorf("Error parsing udp packet: %s", err)
+			level.Error(c.logger).Log("msg", "Error parsing udp packet", "error", err)
```
Comment (matthiasr) [inline main.go]:
```suggestion
			level.Error(c.logger).Log("msg", "Error parsing udp packet", "err", err)
```

File: main.go [hunk 4]
```diff
@@ -137,7 +142,7 @@ func (c *influxDBCollector) parsePointsToSample(points []models.Point) {
 	for _, s := range points {
 		fields, err := s.Fields()
 		if err != nil {
-			log.Errorf("error getting fields from point: %s", err)
+			level.Error(c.logger).Log("msg", "error getting fields from point", "error", err)
```
Comment (matthiasr) [inline main.go]:
```suggestion
			level.Error(c.logger).Log("msg", "error getting fields from point", "err", err)
```

**Comment by bharaththiruveedula** [comment]
@matthiasr  updated the commit  as per your suggestions. Also ran go mod vendor and committed ... still I see the tests faling

**Comment by matthiasr** [comment]
it seems like the ordering in `vendor/modules.txt` is different. which Go version are you using?

**Comment by bharaththiruveedula** [comment]
> it seems like the ordering in `vendor/modules.txt` is different. which Go version are you using?

I am using go 1.12.10

**Comment by matthiasr** [comment]
Please do the `go mod` bits again using Go 1.13, that's the version we are
building with. Sometimes there are small-ish differences between them.

**Comment by bharaththiruveedula** [comment]
@matthiasr I manually rearranged the order to make golint happy... seems like tests are passing :)

**Comment by matthiasr** [comment]
Thanks! There's one more change open (`warn`->`err`), could you commit that please?

**Comment by matthiasr** [comment]
(PS: it's better if you open a branch in your fork to create the PR from – then you can tick the "allow maintainer to make changes" checkbox and I can commit such minor fixes directly to your PR)

**Comment by bharaththiruveedula** [comment]
@matthiasr done!

**Pull request merged**